packages/core-app-api/src/app/AppManager.tsx

```diff
@@ -154,7 +154,53 @@ function useConfigLoader(
     };
   }
 
-  const configReader = ConfigReader.fromConfigs(config.value ?? []);
+  let configReader;
+
+  /**
+   * config.value can be undefined or empty. If it's either, don't bother overriding anything.
+   */
+  if (config.value?.length) {
+    configReader = ConfigReader.fromConfigs(config.value);
+
+    const resolveRelativeUrl = (relativeUrl: string) =>
+      new URL(relativeUrl, document.location.origin).href;
+
+    const getRelativeUrl = (fullUrl: string) => {
+      const url = new URL(fullUrl);
+      return fullUrl.replace(url.origin, '');
+    };
+
+    const getOrigin = (url: string) => new URL(url).origin;
+
+    const appBaseUrl = configReader.getString('app.baseUrl');
+    const backendBaseUrl = configReader.getString('backend.baseUrl');
```

Comment:
On second though using `getString` here stirred up the tests too much, didn't realize the optional gets were guarding agains undefined. I think it's best that we support that so that we don't need the test updates. Should be fairly simple though, we can simply bail if the app URL isn't set, and if it's the only one that's set we just do the app URL transform bit.